src/blockchains/eos/index.ts

```diff
@@ -101,74 +107,88 @@ export function getBalance(string) {
 }
 
 export async function buildPathsFile() {
-    if (Paths) return;
     const tokens = {};
+    const smartTokens = {};
     await Promise.all(converterBlockchainIds.map(async converterBlockchainId => {
         const smartToken = await getSmartToken(converterBlockchainId);
         const smartTokenContract = smartToken.rows[0].smart_contract;
         const smartTokenName = getSymbol(smartToken.rows[0].smart_currency);
         const reservesObject = await getReservesFromCode(converterBlockchainId);
         const reserves = Object.values(reservesObject.rows);
-        tokens[smartTokenContract] = { [smartTokenName]: [converterBlockchainId]};
+        smartTokens[smartTokenContract] = { [smartTokenName]: { [smartTokenName]: converterBlockchainId } };
         reserves.map((reserveObj: Reserve) => {
             const reserveSymbol = getReserveTokenSymbol(reserveObj);
             const existingRecord = tokens[reserveObj.contract];
             if (existingRecord)
-                existingRecord[reserveSymbol].push(converterBlockchainId);
+                existingRecord[reserveSymbol][smartTokenName] = converterBlockchainId;
 
-            tokens[reserveObj.contract] = existingRecord ? existingRecord : { [reserveSymbol]: [converterBlockchainId]};
+            tokens[reserveObj.contract] = existingRecord ? existingRecord : { [reserveSymbol]: { [smartTokenName]: converterBlockchainId } };
         });
     }));
     // eslint-disable-next-line no-console
-    fs.writeFile('./src/blockchains/eos/paths.ts', `export const Paths = ${JSON.stringify(tokens)}`, 'utf8', () => console.log('Done making paths json'));
+    await fs.writeFile('./src/blockchains/eos/paths.ts', `export const Paths = \n{convertibleTokens:${JSON.stringify(tokens)}, \n smartTokens: ${JSON.stringify(smartTokens)}}`, 'utf8', () => console.log('Done making paths json'));
 }
 
 function isFromSmartToken(pair: ConversionPathStep, reserves: string[]) {
-    return (!reserves.includes(pair.fromToken));
+    return (!reserves.includes(Object.values(pair.fromToken)[0]));
 }
 
 function isToSmartToken(pair: ConversionPathStep, reserves: string[]) {
-    return (!reserves.includes(pair.toToken));
+    return (!reserves.includes(Object.values(pair.toToken)[0]));
 }
 
 export async function getPathStepRate(pair: ConversionPathStep, amount: string) {
-    const converterBlockchainId = pair.converterBlockchainId;
-    const reserves = await getReservesFromCode(converterBlockchainId);
+    const toTokenBlockchainId = Object.values(pair.toToken)[0];
```

Comment:
and that's exactly why I didn't want to use objects - object.keys/values iterate through the entire object, allocates extra memory for an array and copies the keys/values into the array.
If we don't access the values by keys, we should just use arrays instead.
Let's discuss.